jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -963,6 +971,20 @@ private static final MethodHandle resolveInvokeDynamic(Class<?> clazz, String na
 		return result;
 	}
 	
+	/*
+	 * Throw NoClassDefFoundError if the cause of TypeNotPresentException is 
+	 * ClassNotFoundException. Otherwise, re-throw TypeNotPresentException.
+	 */
+	private static Throwable noClassDefFoundError(TypeNotPresentException e) {
```

Comment:
Can you add proper javadoc for this method?  Also, to match the naming convention for this kind of method, can you prepend it with `throw` aka `throwNoClassDefFoundError(...)`?

Reply:
fixed.